File: AppGenerateFiles/dbo/Model/Evaluaciones.cs
using CAPA_DATOS;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
namespace DataBaseModel {
   public class Evaluaciones : EntityClass {
       [PrimaryKey(Identity = true)]
       public int? Id { get; set; }
       public DateTime? Fecha { get; set; }
       public ? Hora { get; set; }
       public string? Tipo { get; set; }
       public Double? Porcentaje { get; set; }
       public int? Materia_id { get; set; }
       public int? Seccion_id { get; set; }
       public string? Observaciones { get; set; }
       public DateTime? Created_at { get; set; }
       public DateTime? Updated_at { get; set; }
       public int? Periodo { get; set; }
       public Double? Nota_maxima { get; set; }
       [ManyToOne(TableName = "Materias", KeyColumn = "Id", ForeignKeyColumn = "Materia_id")]
       public Materias? Materias { get; set; }
       [ManyToOne(TableName = "Secciones", KeyColumn = "Id", ForeignKeyColumn = "Seccion_id")]
       public Secciones? Secciones { get; set; }
       [OneToMany(TableName = "Calificaciones", KeyColumn = "Id", ForeignKeyColumn = "Evaluacion_id")]
       public List<Calificaciones>? Calificaciones { get; set; }
   }
}

File: UI/wwwroot/Model/Evaluaciones.js
//@ts-check
// @ts-ignore
import { EntityClass } from '../WDevCore/WModules/EntityClass.js';
import { Calificaciones } from './Calificaciones.js';
import { Materias } from './Materias.js';
import { Secciones } from './Secciones.js';
class Evaluaciones extends EntityClass {
   /** @param {Partial<Evaluaciones>} [props] */
   constructor(props) {
       super(props, 'EntityDbo');
       for (const prop in props) {
           this[prop] = props[prop];
       }
   }
   /**@type {Number}*/ Id;
   /**@type {String}*/ Fecha;
   /**@type {TimeRanges}*/ Hora;
   /**@type {String}*/ Tipo;
   /**@type {Number}*/ Porcentaje;
   /**@type {String}*/ Observaciones;
   /**@type {Date}*/ Created_at;
   /**@type {Date}*/ Updated_at;
   /**@type {Number}*/ Periodo;
   /**@type {Number}*/ Nota_maxima;
   /**@type {Materias} ManyToOne*/ Materias;
   /**@type {Secciones} ManyToOne*/ Secciones;
   /**@type {Array<Calificaciones>} OneToMany*/ Calificaciones;
}
export { Evaluaciones };

